winconf.pri

```diff
@@ -1,68 +1,82 @@
-# Point this to the libtorrent lib folder
-LIBS += $$quote(-LC:/qBittorrent/RC_0_16/bin/path-according-to-the-build-options-chosen)
-# Point this to the zlib lib folder
-LIBS += $$quote(-LC:/qBittorrent/Zlib/lib)
-# Point this to the openssl lib folder
-LIBS += $$quote(-LC:/qBittorrent/openssl/lib)
-
-# BOOST DEFINES
-DEFINES += BOOST_ALL_NO_LIB
 DEFINES += BOOST_ASIO_HASH_MAP_BUCKETS=1021
-# Disable this if building against libtorrent 1.1.x (RC_1_1)
-DEFINES += BOOST_ASIO_SEPARATE_COMPILATION
 # After 1.55 some Windows users reported regular UI freezes.
 # This makes ASIO use the pre-1.56 way of doing things. See issue #2003
 DEFINES += BOOST_ASIO_DISABLE_CONNECTEX
-# Boost 1.60+ defaults to Vista+ support. The define below enables XP support again.
-DEFINES += BOOST_USE_WINAPI_VERSION=0x0501
 DEFINES += BOOST_EXCEPTION_DISABLE
-DEFINES += BOOST_SYSTEM_STATIC_LINK=1
 
-# LIBTORRENT DEFINES
 DEFINES += TORRENT_USE_OPENSSL
 DEFINES += TORRENT_DISABLE_GEO_IP
 DEFINES += TORRENT_DISABLE_RESOLVE_COUNTRIES
 
-# QBT DEFINES
 DEFINES += UNICODE
 DEFINES += _UNICODE
 DEFINES += WIN32
 DEFINES += _WIN32
 DEFINES += WIN32_LEAN_AND_MEAN
-DEFINES += NTDDI_VERSION=0x05010000
-DEFINES += _WIN32_WINNT=0x0501
-DEFINES += _WIN32_IE=0x0501
 DEFINES += _CRT_SECURE_NO_DEPRECATE
 DEFINES += _SCL_SECURE_NO_DEPRECATE
 DEFINES += __USE_W32_SOCKETS
 DEFINES += _FILE_OFFSET_BITS=64
 DEFINES += NOMINMAX
 
 CONFIG(debug, debug|release) {
-  DEFINES += TORRENT_DEBUG
-} else {
-  DEFINES += NDEBUG
+    DEFINES += TORRENT_DEBUG
+}
+else {
+    DEFINES += NDEBUG
 }
-
-# Enable backtrace support
-CONFIG += strace_win
 
 win32-g++* {
-    include(winconf-mingw.pri)
+    CONFIG(debug, debug|release) {
+        # Make sure binary is not relocatable, otherwise debugging will fail
+        QMAKE_LFLAGS -= -Wl,--dynamicbase
+    }
+
+    RC_FILE = qbittorrent_mingw.rc
+    
+    LIBS += libadvapi32 libshell32 libuser32 libole32 libwsock32 libws2_32 libpowrprof
+}
+else {
+    CONFIG -= embed_manifest_exe
+    QMAKE_LFLAGS += "/MANIFEST:EMBED /MANIFESTINPUT:$$quote($${PWD}/src/qbittorrent.exe.manifest) /STACK:0x800000"
+    QMAKE_LFLAGS_RELEASE += "/OPT:REF /OPT:ICF"
+
+    RC_FILE = qbittorrent.rc
+    
+    LIBS += advapi32.lib shell32.lib crypt32.lib User32.lib ole32.lib PowrProf.lib
+}
+
+# See an example build configuration in "conf.pri.windows"
+exists(conf.pri) {
+    include(conf.pri)
 }
 else {
-    include(winconf-msvc.pri)
+    error("'conf.pri' does not exist. See an example configuration in 'conf.pri.windows'.")
+}
+
+# Stack trace support can be enabled in 'conf.pri'
+strace_win {
+    win32-g++* {
+        contains(QMAKE_HOST.arch, x86) {
+            # i686 arch requires frame pointer preservation
+            QMAKE_CXXFLAGS_RELEASE += -fno-omit-frame-pointer
+            QMAKE_CXXFLAGS_DEBUG += -fno-omit-frame-pointer
+        }
+
+        QMAKE_LFLAGS += -Wl,--export-all-symbols
+
+        LIBS += libdbghelp
+    }
+    else {
```

Comment:
would use `win32-msvc*` here...

Overall it's pretty good, if you don't wish to change anymore, I'll give my approval.